### PR TITLE
update train scripts

### PR DESCRIPTION
**Pull request by aihao2000**
# What does this PR do?
- tensorboard log 
   When the validation prompt is the same, the validation will be overwritten, so I personally recommend using a unique id.
- load_dataset_num_proc
   Multiple processes can significantly increase the speed of load_dataset calculation cache.
- load_dataset data_dir
   Supports custom local datasets. Just like https://github.com/huggingface/diffusers/pull/5747#event-10922633415

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md)?
- [ ] Did you read our [philosophy doc](https://github.com/huggingface/diffusers/blob/main/PHILOSOPHY.md) (important for complex PRs)?
- [ ] Was this discussed/approved via a GitHub issue or the [forum](https://discuss.huggingface.co/c/discussion-related-to-httpsgithubcomhuggingfacediffusers/63)? Please add a link to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/diffusers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/diffusers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@sayakpaul and @patrickvonplaten

### CODE REVIEW

File: examples/controlnet/train_controlnet.py [hunk 1]
```diff
@@ -485,6 +485,9 @@ def parse_args(input_args=None):
         default="text",
         help="The column of the dataset containing a caption or a list of captions.",
     )
+    parser.add_argument(
+        "--load_dataset_num_proc", type=int, default=None
```
Comment (sayakpaul) [inline examples/controlnet/train_controlnet.py:489]: Please add a helper as well.

File: examples/controlnet/train_controlnet_sdxl.py [hunk 2]
```diff
@@ -537,7 +542,7 @@ def parse_args(input_args=None):
     parser.add_argument(
         "--num_validation_images",
         type=int,
-        default=4,
+        default=1,
```
Comment (sayakpaul) [inline examples/controlnet/train_controlnet_sdxl.py:545]: Why?

Comment (aihao2000) [reply examples/controlnet/train_controlnet_sdxl.py]: @sayakpaul Already restored it.

File: examples/controlnet/train_controlnet.py (hunk 1 above)
Comment (aihao2000) [reply examples/controlnet/train_controlnet.py]: @sayakpaul added

**Pull request closed without merging**